backend/server.js

```diff
@@ -538,6 +529,253 @@ app.post('/subscribeEmail', wrap(async (req, res) => {
 }));
 
 
+// finds the user with the login key that's been requested
+// if the user doesn't exist, return
+async function findMatchingUser(requestLoginKey) {
+  // Loop over the db
+  const users = await database.get('users');
+  if (!users) {
+    return null;
+  }
+
+  // Loop over all the users
+  for (const user of users) {
+    for (const loginKey of user.loginKeys) {
+      if (requestLoginKey === loginKey) {
+        return user;
+      }
+    }
+  }
+
+  return null;
+}
+
+
+// sends data to the database in the backend
+async function sendUserHelper(req, res) {
+  // Don't cache this endpoint.
+  res.setHeader('Cache-Control', 'no-cache, no-store');
+
+
+  // if there's no body in the request, well, we'll crash, so let's not
+  if (!req.body || !req.body.loginKey) {
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return null;
+  }
+
+  // Checks checks checks
+  // Make sure the login key is valid
+  if (typeof req.body.loginKey !== 'string' || req.body.loginKey.length !== 100) {
+    macros.log('Invalid login key', req.body.loginKey);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return null;
+  }
+
+  const senderId = req.body.senderId;
+
+  // Make sure the sender id is valid
+  if (senderId && (typeof senderId !== 'string' || senderId.length !== 16 || !macros.isNumeric(senderId))) {
+    macros.log('Invalid senderId', req.body, senderId);
+    res.send(JSON.stringify({
+      error: 'Error.',
+    }));
+    return null;
+  }
+
+
+  // If the client specified a specific senderId, lookup that specific user.
+  // if not, we have to loop over all the users's to find a matching loginKey
+  if (senderId) {
+    // first confirm the loginkey is legitimate
+    const user = await database.get(`/users/${senderId}`);
+    if (user && !user.loginKeys.includes(req.body.loginKey)) {
+      macros.log('loginKey not within the sender\'s entry');
+      res.send(JSON.stringify({
+        error: 'Error, invalid loginkey on senderId.',
+      }));
+      return null;
+    }
+    return user;
+  }
+  macros.log('Really invalid senderID', req.body, senderId);
+  res.send(JSON.stringify({
+    error: 'Error.',
+  }));
+  return null;
+  // to make sure i'm not shooting myself in the foot :)
+}
+
+
+app.post('/addSection', wrap(async (req, res) => {
+  const userObject = await sendUserHelper(req, res);
+
+  userObject.watchingSections.push(req.body.info);
+
+  await database.set(`/users/${req.body.senderId}`, userObject);
+  macros.log('sending done.');
+
+  const sectionInfo = req.body.sectionInfo;
+
+  notifyer.sendFBNotification(req.body.senderId, `Thanks for signing up for a section in ${sectionInfo.subject} ${sectionInfo.classId} (CRN: ${sectionInfo.crn}). Check it out at https://searchneu.com/${sectionInfo.termId}/${sectionInfo.subject}${sectionInfo.classId} !`);
```

Comment:
We don't need to link back to the site when we send this msg - can you remove this part? 

```suggestion
  notifyer.sendFBNotification(req.body.senderId, `Thanks for signing up for a section in ${sectionInfo.subject} ${sectionInfo.classId} (CRN: ${sectionInfo.crn}).`);
```